heroku/data_source_heroku_app.go

```diff
@@ -0,0 +1,106 @@
+package heroku
+
+import (
+	"context"
+	"github.com/cyberdelia/heroku-go/v3"
+	"github.com/hashicorp/terraform/helper/schema"
+	"time"
+)
+
+func dataSourceHerokuApp() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHerokuAppRead,
+		Schema: map[string]*schema.Schema{
```

Comment:
Please be sure these are documented somewhere.

Reply:
I don't have experience with data sources, but I wonder if it would be useful to re-use the existing app resource schema here? Something like `Schema: resourceHerokuApp().Schema`?

For that matter, could you use `resourceHerokuAppRead` as the reader here to also re-use all of that logic?

Reply:
@bernerdschaefer I'll try to DRY this up later on, but I'm not sure how much I can do regarding schema reuse. The schema actually is subtly different - for instance, elements tend to be "required" in the resource schema, but "computed" in the datasource schema. 

I'm using aws resources and datasources for reference (this is also my first experience writing datasources, or any terraform go code).

Reply:
Ah, interesting. The repetition in read bothers me more than in the schema since there's rather a lot going on, but neither bothers me so badly that we couldn't ship this as a first pass.

Reply:
@bernerdschaefer I've refactored a bunch of the logic around retrieving data for an app, so that I can reuse stuff from `resource_heroku_app` inside of `datasource_heroku_app`. 

@Xe I've added docs on `datasource_heroku_app`, using the existing heroku_app resource docs plus some aws datasource docs as a template.

It also now retrieves a bunch more stuff than the first pass did - buildpacks, config vars, organization info.

I still need to manually test organization and space functionality (so far I've only been testing with a common runtime app), and I need to write some tests.